packages/client/hmi-client/src/components/workflow/scenario-templates/policy-design/policy-design-scenario.ts

```diff
@@ -0,0 +1,244 @@
+		};
+	}
+
+	isValid(): boolean {
+		return !!this.workflowName && !!this.modelId && !!this.modelConfigId && !!this.interventionPolicyId;
+	}
+
+	async createWorkflow() {
+		let wf = new workflowService.WorkflowWrapper();
+		wf.setWorkflowName(this.workflowName);
+		wf.setWorkflowScenario(this.toJSON());
+		wf = await this.addAllNodes(wf);
+		wf.runDagreLayout();
+		return wf.dump();
+	}
+
+	getModelId() {
+		return this.modelId;
+	}
+
+	setModelId(modelId: string) {
+		this.modelId = modelId;
+	}
+
+	getModelConfigId() {
+		return this.modelConfigId;
+	}
+
+	setModelConfigId(modelConfigId: string) {
+		this.modelConfigId = modelConfigId;
+	}
+
+	getInterventionPolicyId() {
+		return this.interventionPolicyId;
+	}
+
+	async setInterventionPolicyId(interventionPolicyId: string, modelConfiguration?: ModelConfiguration) {
+		this.interventionPolicyId = interventionPolicyId;
+		// Update optimize state if able:
+		if (modelConfiguration) {
+			const state = this.optimizeState;
+			const interventionPolicy = await getInterventionPolicyById(interventionPolicyId);
+			if (interventionPolicy) {
+				this.optimizeState.interventionPolicyGroups = setInterventionPolicyGroups(
+					state,
+					interventionPolicy,
+					modelConfiguration
+				).interventionPolicyGroups;
+			}
+		}
+	}
+
+	getDatasetId() {
+		return this.datasetId;
+	}
+
+	setDatasetId(datasetId: string) {
+		this.datasetId = datasetId;
+	}
+
+	getOptimizeState() {
+		return this.optimizeState;
+	}
+
+	updateInterventionPolicyGroupForm(index: number, config: InterventionPolicyGroupForm) {
+		this.optimizeState.interventionPolicyGroups[index] = config;
+	}
+
+	getOptimizeOutputSettings() {
+		return this.optimizeState.chartSettings?.flatMap((chartSettings) => chartSettings.selectedVariables) ?? [];
+	}
+
+	setOptimizeOutputSettings(selectedVariables: string[]) {
+		let optimizeChartSettings: ChartSetting[] = [];
+		optimizeChartSettings = updateChartSettingsBySelectedVariables(
+			optimizeChartSettings,
+			ChartSettingType.VARIABLE,
+			selectedVariables
+		);
+		this.optimizeState.chartSettings = optimizeChartSettings;
+	}
+
+	private async addAllNodes(wf: workflowService.WorkflowWrapper): Promise<workflowService.WorkflowWrapper> {
+		// Add Default Nodes (not calibrate + dataset):
+		const POSITION = { x: 0, y: 0 };
+		const SIZE = { size: OperatorNodeSize.medium };
+		const modelConfig = await getModelConfigurationById(this.modelConfigId);
+		const interventionPolicy = await getInterventionPolicyById(this.interventionPolicyId);
+
+		const modelNode = wf.addNode(ModelOp, POSITION, SIZE);
+
+		const modelConfigNode = wf.addNode(ModelConfigOp, POSITION, SIZE);
+
+		const interventionNode = wf.addNode(InterventionOp, POSITION, SIZE);
+
+		const optimizeNode = wf.addNode(OptimizeOp, POSITION, SIZE);
+
+		// Add Edges:
+		// Model Config Input
+		wf.addEdge(modelNode.id, modelNode.outputs[0].id, modelConfigNode.id, modelConfigNode.inputs[0].id, [
+			{ x: 0, y: 0 },
```

Comment:
```suggestion
			POSITION,
```

?

Same for others below

Reply:
Thanks, there was a fair bit of copy paste in this rather ugly section I will clean that up